### ISSUE TITLE
vm-insert panic when upgrading to v1.34.0

### ISSUE THREAD

**Issue by hekmon** [VictoriaMetrics/VictoriaMetrics#339]
**Describe the bug**
It seems to be linked to the `projectID` on the tenant URL (not sure, just guessing based on the stack trace). We are only using prometheus but I think that all insert are affected.

**To Reproduce**
Start insert module
--> OK
Insert any metrics using an `accountID` and a `projectID`:
`http://<vminsert>:8480/insert/<accountID>:<projectID>/<suffix>`
--> panic

**Expected behavior**
No panic/metrics inserted.

**Logs / Stack trace**
```
Feb 26 14:09:33 vmc-insert-02 systemd[1]: Started Victoria Metrics - cluster version: insert module.
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:11        build version: vminsert-20200224-225226-tags-v1.34.0-cluster-0-gca28a3e8
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:12        command line flags
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "enableTCP6" = "false"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "envflag.enable" = "true"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "fs.disableMmap" = "false"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "graphiteListenAddr" = ""
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "http.disableResponseCompression" = "false"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "httpListenAddr" = "0.0.0.0:8480"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "import.maxLineLen" = "104857600"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "influxMeasurementFieldSeparator" = "_"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "influxSkipSingleField" = "false"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "insert.maxQueueDuration" = "1m0s"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "loggerFormat" = "default"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "loggerLevel" = "INFO"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "loggerOutput" = "stderr"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "maxConcurrentInserts" = "8"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "maxInsertRequestSize" = "33554432"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "maxLabelsPerTimeseries" = "30"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "memory.allowedPercent" = "60"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "opentsdbHTTPListenAddr" = ""
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "opentsdbListenAddr" = ""
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "opentsdbhttp.maxInsertRequestSize" = "33554432"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "rpc.disableCompression" = "false"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "storageNode" = "vmc-storage-01:8400,vmc-storage-02:8400"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/lib/logger/flag.go:20        flag "version" = "false"
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.010Z        info        VictoriaMetrics@/app/vminsert/main.go:55        initializing netstorage for storageNodes [vmc-storage-01:8400 vmc-storage-02:8400]...
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.017Z        info        VictoriaMetrics@/lib/memory/memory.go:33        limiting caches to 1253833113 bytes, leaving 835888743 bytes to the OS according to -memory.allowedPercent=60
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.017Z        info        VictoriaMetrics@/app/vminsert/main.go:61        successfully initialized netstorage in 0.007 seconds
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.017Z        info        VictoriaMetrics@/lib/httpserver/httpserver.go:48        starting http server at http://0.0.0.0:8480/
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.017Z        info        VictoriaMetrics@/lib/httpserver/httpserver.go:49        pprof handlers are exposed at http://0.0.0.0:8480/debug/pprof/
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: 2020-02-26T14:09:33.670Z        error        net/http/server.go:3054        http: panic serving 127.0.0.1:60016: runtime error: index out of range [0] with length 0
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: goroutine 33 [running]:
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: net/http.(*conn).serve.func1(0xc0000a1680)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         net/http/server.go:1767 +0x139
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: panic(0x9e35c0, 0xc000098e80)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         runtime/panic.go:679 +0x1b2
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/netstorage.(*InsertCtx).WriteDataPointExt(0xc0000d8fc0, 0xc00009d3b8, 0x0, 0xc0000bc1c0, 0xd3, 0xe0, 0x17081d4bd33, 0x414f638280000000, 0xd3, 0xe0)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/app/vminsert/netstorage/insert_ctx.go:125 +0x27e
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.insertRows(0xc00009d3b8, 0xc0000cf000, 0x44, 0x80, 0x0, 0x0)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/app/vminsert/promremotewrite/request_handler.go:44 +0x14d
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.InsertHandler.func1.1(0xc0000cf000, 0x44, 0x80, 0x0, 0x20)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/app/vminsert/promremotewrite/request_handler.go:25 +0x48
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/promremotewrite.ParseStream(0xc0000dcf00, 0xc0000e7890, 0x0, 0x0)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/lib/protoparser/promremotewrite/streamparser.go:28 +0xdb
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.InsertHandler.func1(0xc000074ba0, 0xc0000e78e8)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/app/vminsert/promremotewrite/request_handler.go:24 +0x51
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/lib/writeconcurrencylimiter.Do(0xc0000e79c8, 0xc00009d3b8, 0x4)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/lib/writeconcurrencylimiter/concurrencylimiter.go:36 +0x2fc
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.InsertHandler(0xc00009d3b8, 0xc0000dcf00, 0xc00009d3b8, 0x0)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/app/vminsert/promremotewrite/request_handler.go:23 +0x5c
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: main.requestHandler(0xac2320, 0xc0001a4060, 0xc0000dcf00, 0xc0001a4000)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/app/vminsert/main.go:135 +0x1cc
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xac2320, 0xc0001a4060, 0xc0000dcf00, 0xa3f820)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/lib/httpserver/httpserver.go:163 +0x185
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1(0xac2320, 0xc0001a4060, 0xc0000dcf00)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         github.com/VictoriaMetrics/VictoriaMetrics@/lib/httpserver/httpserver.go:117 +0x8e
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: net/http.HandlerFunc.ServeHTTP(0xc000158d20, 0xac2720, 0xc0001940e0, 0xc0000dcf00)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         net/http/server.go:2007 +0x44
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: net/http.serverHandler.ServeHTTP(0xc000194000, 0xac2720, 0xc0001940e0, 0xc0000dcf00)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         net/http/server.go:2802 +0xa4
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: net/http.(*conn).serve(0xc0000a1680, 0xac35a0, 0xc000072d40)
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         net/http/server.go:1890 +0x875
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]: created by net/http.(*Server).Serve
Feb 26 14:09:33 vmc-insert-02 vmc-insert[5196]:         net/http/server.go:2928 +0x384
```

**Version**

```
vminsert-20200224-225226-tags-v1.34.0-cluster-0-gca28a3e8
```

**Used command-line flags**
Env vars used, see the logs for values.

**Additional context**
As `projectID` is an undocumented feature, I am not sure if it is a bug or an hidden feature removal.

**Comment by valyala** [comment]
Confirmed the bug. It should be fixed in the commit 1286cead759903084fd9f4c25051c3e995f8bbbb . @hekmon , could you build `vminsert` from this commit and verify whether it works as expected? See [these docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/README.md#building-from-sources) about building cluster components from sources.

**Comment by valyala** [comment]
The bug should be fixed in [v1.34.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue.

**Comment by valyala** [comment]
> As projectID is an undocumented feature, I am not sure if it is a bug or an hidden feature removal.

This was a bug. As for `projectID`, there are no plans for its removal, so feel free relying on it. It is undocumented because single-level tenants (aka accountID) is enough for the majority of users :)

**Comment by hekmon** [comment]
> Confirmed the bug. It should be fixed in the commit [1286cea](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/1286cead759903084fd9f4c25051c3e995f8bbbb) . @hekmon , could you build `vminsert` from this commit and verify whether it works as expected? See [these docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/README.md#building-from-sources) about building cluster components from sources.

Sorry did not have the time to test it until now :(

> The bug should be fixed in [v1.34.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue.

I can confirm it is :) Thanks !

> This was a bug. As for `projectID`, there are no plans for its removal, so feel free relying on it. It is undocumented because single-level tenants (aka accountID) is enough for the majority of users :)

That's great use as we are heavily using the `projectID` for some `accountID`s :+1:
